### ISSUE TITLE
Brace rules confirmation

### ISSUE THREAD

**Issue by craigbilner** [netguru/eslint-plugin-ember#26]
Should computed keys be braced with a preference for longer prefixes?

i.e.

```js
computed('foo.bar.{baz,bash}', 'foo.baz')
```

is preferred to

```js
computed('foo.{baz,bar.baz,bar.bash}')
```

**Comment by netes** [comment]
IMO, first variant is much more readable.

**Comment by michalsnik** [comment]
I agree with @netes, but it looks like our current rule wants the second option, which might not be a good thing in some cases. I wouldn't give it a big priority though, it's worth checking if we can improve this rule to be more `intelligent`.

**Comment by clcuevas** [comment]
Are there any updates to improving this rule?

**Comment by michalsnik** [comment]
Not yet @clcuevas I'm giving it second thought right now

**Comment by michalsnik** [comment]
I tried to come up with something, but there seem to be so many cases, that it's not so easy to grasp. What I would suggest is to gather as many different cases here in this conversation to define logic behind. Then I can prepare test scenarios and start implementing it.

**Comment by clcuevas** [comment]
Hi @michalsnik Aside from the original poster's example, another scenario that I ran into is when computed properties may be implemented like this:

```
someComputedProperty: on('didInsertElement', observer('foo.{bar,baz}', function() {
  // code goes here
});
```

There have been some use cases in where I need to create computed properties this way and I believe (I can double check on this) from what I remember, the current rule does not capture this scenario. So, essentially, I can write the above this way as well and the rule will not detect it:

```
someComputedProperty: on('didInsertElement', observer('foo.bar', 'foo.baz', function() {
  // code goes here
});
```

I've actually been meaning to re-visit this rule and see how it can be improved when I get the chance. If there is any way I can help, please let me know.

**Comment by michalsnik** [comment]
@clcuevas ah you see, it should work, but for `computed`. We're detecting computed properties but not observers.
